packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -154,24 +210,42 @@ const VisitNotesForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patie
           },
         ],
         encounterType: encounterTypeUuid,
-        form: formConceptUuid,
         obs: obs,
       };
 
       const abortController = new AbortController();
       saveVisitNote(abortController, visitNotePayload)
         .then((response) => {
           if (response.status === 201) {
-            closeWorkspace();
-
-            showToast({
-              critical: true,
-              description: t('visitNoteNowVisible', 'It is now visible on the Encounters page'),
-              kind: 'success',
-              title: t('visitNoteSaved', 'Visit note saved'),
+            combinedDiagnosis.map((diagnosis, position: number) => {
+              let diagnosisPayload: DiagnosisPayload = {
+                encounter: response.data.uuid,
+                patient: patientUuid,
+                condition: null,
+                diagnosis: {
+                  coded: diagnosis.diagnosis.coded,
+                },
+                certainty: diagnosis.certainty,
+                rank: diagnosis.rank,
+              };
+              savePatientDiagnoses(abortController, diagnosisPayload).then((response) => {
+                if (response.status == 201) {
+                  if (position > -1) {
+                    combinedDiagnosis.splice(position, 1);
+                  }
+                  if (combinedDiagnosis.length == 0) {
```

Comment:
I'd think that modifying `combinedDiagnosis` (which should be `combinedDiagnoses`) while iterating is weird and potentially could cause thrashing. Could we just swap this with:

```ts
if (position === combinedDiagnoses.length - 1) {
```

Then inside the `if` block, we can "clear" the array (e.g., `combinedDiagnoses.length = 0` if that's necessary... Element-wise splicing is a relatively expensive operation